### ISSUE TITLE
Error: Please provide RewritingOptions.AssemblyPaths

### ISSUE THREAD

**Issue by luisgepeto** [microsoft/coyote#380]
I am hitting the following exception when running the `coyote rewrite .\rewrite.coyote.json` command.
According to the documentation the [`Assemblies` property is an optional list](https://microsoft.github.io/coyote/#get-started/using-coyote/#configuration). However this is the content of my `rewrite.coyote.json` file:

```json
{
  "AssembliesPath": ".",
  "OutputPath": "./coyote_rewrite"
}
```

Executing the aforementioned command results in the following error:
> Error: Please provide RewritingOptions.AssemblyPaths

This exception seems to be thrown by [this line in the RewritingOptions.cs class](https://github.com/microsoft/coyote/blob/8d0fe88d1135e020c8a53a39b2f2fbbc9f734689/Source/Test/Rewriting/RewritingOptions.cs#L247). This is due to the json not being correctly parsed as per documentation inside the [`RewritingOptions.ParseFromJSON` function](https://github.com/microsoft/coyote/blob/8d0fe88d1135e020c8a53a39b2f2fbbc9f734689/Source/Test/Rewriting/RewritingOptions.cs#L158)

This issue can by bypassed by setting the `rewrite.coyote.json` file contents to be:
```json
{
  "AssembliesPath": ".",
  "OutputPath": "./coyote_rewrite",
  "Assemblies": ["."]
}
```

Either way, the documentation or the code behavior needs to be updated

**Comment by pdeligia** [comment]
Thanks for catching this issue @luisgepeto! I have not hit this before (nor been reported), so really helpful to know this can happen. Give us some time and we will address this, but glad there is an easy workaround so you are not blocked.

**Comment by luisgepeto** [comment]
Thanks @pdeligia ! Actually, setting assemblies to `"Assemblies": ["."]` is also not working for me. I need to explicitly set the dll names to be rewritten.

**Comment by pdeligia** [comment]
That is correct @luisgepeto, actually anything besides explicitly setting the DLLs to rewrite is not supported today (so things like pattern matching, `*` and `.` are not supported, at list not right now). I just updated the rewriting documentation on the website to clarify this (and address this issue), and I also added a recommendation on what to rewrite based on our internal discussion. Closing this issue for now!
